src/cdk/tree/navigator.ts

```diff
@@ -0,0 +1,304 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {FocusableOption} from '@angular/cdk/a11y';
+import {LEFT_ARROW, RIGHT_ARROW, UP_ARROW, DOWN_ARROW, HOME, END, ENTER, TAB} from '@angular/cdk/keycodes';
+import {Directive} from '@angular/core';
+import {Subject} from 'rxjs';
+import {TreeControl} from './control/tree-control';
+
+const NUMPAD_MULTIPLY_KEYCODE = 106;
+
+/**
+ * Keyboard navigator interface.
+ */
+// TODO(tinayuangao): Move to a common place
+export interface CdkNavigator<T> {
+  insert(index: number, data: T, node: FocusableOption, parentInfo?: any);
```

Comment:
Isn't the `parentInfo` here of type `T`? It seems like it is in the usage below. Also the parameter here is called `parentInfo`, but below it's `parentData`.